migrations: adds necessary migrations for `settings.network`

etungsten opened the pull request:
<!--
Tips:
- Please read CONTRIBUTING.md to understand our process and our requests for PRs.
- Please file an issue before creating a PR so we can discuss the change and confirm it's not already being worked on.
-->

**Issue number:**
N/A


**Description of changes:**

This adds the migration related to https://github.com/bottlerocket-os/bottlerocket/pull/1204 and https://github.com/bottlerocket-os/bottlerocket/pull/1231.

```
Author: Erikson Tung <etung@amazon.com>
Date:   Tue Jan 5 17:45:43 2021 -0800

    migrations: adds necessary migrations for `settings.network`
    
    Adds new migrations for migrating new settings and affected service
    settings. We also added new service settings for docker and
    host-containerd to incorporate restart commands for those services.

```


**Testing done:**

Built a custom TUF repository with a v1.0.5 aws-k8s-1.18 image.
Launched a v1.0.4 aws-k8s-1.18 host with a custom root.json and upgraded it to v1.0.5.
The instance came up fine and is able to run pods. Checked the settings and they got migrated as expected.
Rolled back the instance to v1.0.4, the host came back fine and is able to run pods. Backwards migration run successfully, all settings expected to be migrated got migrated.


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

tjkirch reviewed (commented):
This looks fine, but won't we also need migrations to change the configuration-files lists for containerd, host-containerd, docker, and kubernetes to add proxy-env?  I think you can do it with ReplaceListMigration, similar to the sysctl-subcommand migration.

[edit] I missed #1231 - also need to remove the restart commands for containerd and kubernetes, and the whole services.docker and services.host-containerd trees, I think.

Would you please test on an instance with a real repo?  It's easier to do now with pubsys.

etungsten commented:
New migrations to address https://github.com/bottlerocket-os/bottlerocket/pull/1258#pullrequestreview-562052413 are pending changes in https://github.com/bottlerocket-os/bottlerocket/pull/1262 to be merged.

etungsten committed:
migrations: adds necessary migrations for `settings.network`

Adds new migrations for migrating new settings and affected service
settings. We also added new service settings for docker and
host-containerd to incorporate restart commands for those services.

etungsten commented:
Push above adds two new migrations for migrating the new `restart-commands` and `configuration-file` settings.

Updated PR description with new testing.

zmrow reviewed (approved):
:safety_vest:

tjkirch commented on Release.toml:
```diff
@@ -12,4 +12,11 @@ version = "1.0.4"
+    "migrate_v1.0.5_sysctl-subcommand.lz4",
+    "migrate_v1.0.5_add-user-data.lz4",
+    "migrate_v1.0.5_add-network-settings.lz4",
+    "migrate_v1.0.5_replace-service-settings.lz4",
+    "migrate_v1.0.5_add-services.lz4"
```
I know it's a nitpick, but these names are effectively in a single namespace forever, and this is fairly generic without any reference to which services or the reason for updating/adding services; can we make them a little more specific?  Like "add-proxy-restart" and "add-proxy-services" or something?  They're still fundamentally tied to the proxy feature, given that they add proxy-env, not just restarts.

etungsten commented:
Push above changes the name of the migrations to address https://github.com/bottlerocket-os/bottlerocket/pull/1258#discussion_r553528872
